.github/workflows/ci.yaml

```diff
@@ -0,0 +1,60 @@
+name: CI
+
+on: pull_request
+
+jobs:
+  ci:
+    runs-on: ubuntu-latest
+
+    permissions:
+      contents: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.head_ref }}
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: "20"
+
+      - name: Enable Corepack
+        run: corepack enable
+
+      - name: Install dependencies
+        run: yarn install --immutable
```

Comment:
On main it looks like we [cache dependencies](https://github.com/hashicorp/vault/blob/main/.github/workflows/ci.yml#L275-L276). This action is probably not running as frequently to maybe not worth it, but wanted to point that out.

Reply:
Yep - that makes sense! Something that could always be added later if the dependencies grow here